### DISCUSSION TITLE
what are the rules to be applied while migrating php7.3 to php8.0 ?

### DISCUSSION THREAD

**Discussion by bala686** [rectorphp/rector#7902] [Q&A] [answered]
In the below link 420 rules are mentioned, if we need to apply all the rules which is in the link for migrating php7.3 to 
php 8.0?
Link:
https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#arguments
what are the rules have to be applied for entire migrate process?

**Comment by ian-zunderdorp** [comment, marked as answer]
You can use the [LevelSetList](https://github.com/rectorphp/rector-src/blob/main/packages/Set/ValueObject/LevelSetList.php) instead of configuring individual rules :)

**Reply by 7thstorm** [reply]
The issue I'm facing with this is that if I use levelSetList to UP_TO_PHP_82, it will look for issues in all versions, even before ver 7.3. I want to limit from 7.3 to 8.2, because I know the code works on 7.3, so I want rector to look ahead. I have opened a separate discussion on this point

**Reply by ian-zunderdorp** [reply]
You can use the [SetList](https://github.com/rectorphp/rector-src/blob/main/packages/Set/ValueObject/SetList.php#L12) which only contain the rules to upgrade from previous PHP version to the one specified (so PHP_74 only applies 7.3 -> 7.4 changes)

**Reply by 7thstorm** [reply]
Thanks, But since I'm at 7.4, and I want to go to 8.2, are you saying I have to do it in stages?
i.e 8.0 then 8.1 and finally 8.2??

**Reply by ian-zunderdorp** [reply]
You can import multiple sets simultaneously.
